server/main.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"net/url"
+	"os"
+	"strconv"
+
+	"github.com/vertexcover-io/locatr"
+)
+
+var clientAndLocatrs = make(map[string]locatr.LocatrInterface)
+
+func createLocatrOptions(message incomingMessage) locatr.BaseLocatrOptions {
+	opts := locatr.BaseLocatrOptions{}
+	llmConfig := message.Settings.LlmSettings
+
+	if llmConfig.ReRankerApiKey != "" {
+		opts.ReRankClient = locatr.NewCohereClient(llmConfig.ReRankerApiKey)
+	}
+
+	opts.LogConfig = locatr.LogConfig{Level: locatr.Debug}
+
+	opts.CachePath = message.Settings.CachePath
+	opts.UseCache = message.Settings.UseCache
+
+	opts.ResultsFilePath = message.Settings.ResultsFilePath
+
+	llmClient, _ := locatr.NewLlmClient(
+		locatr.LlmProvider(llmConfig.LlmProvider),
+		llmConfig.ModelName,
+		llmConfig.LlmApiKey,
+	)
+	opts.LlmClient = llmClient
+
+	return opts
+}
+
+func handleLocatrRequest(message incomingMessage) (string, error) {
+	baseLocatr, ok := clientAndLocatrs[message.ClientId]
+	if !ok {
+		return "", fmt.Errorf("Client not instianciated of id: %s", message.ClientId)
```

Comment:
better to create at least an `ErrClientNotInstantiated` otherwise `errors.Is` interface isn't able to capture the type of error.